### ISSUE TITLE
Can't output multivar from module

### ISSUE THREAD

**Issue by publysher** [hashicorp/terraform#845]
In my module, I'm trying to create multiple resources using the `count` parameter. However, it seems to be impossible to output details of the resources I've created. 

For example, in my module I have something like:

```
resource "digitalocean_droplet" "node" {
    count = "${var.servers}"
    ....
}
```

I've tried various ways to export the IP addresses of these nodes:

```
output "ip" {
    value = "${digitalocean_droplet.node.ipv4_address}"
}
```

results in

```
Resource 'digitalocean_droplet.node' not found for variable 'digitalocean_droplet.node.ipv4_address'
```

`value = "${digitalocean_droplet.node.*.ipv4_address}"` results in `output 'ipv4_address': multi-variable must be in a slice`.

`value = ["${digitalocean_droplet.node.*.ipv4_address}"]` parses OK, but results in a string containing only the first ip address.

`value="${join(",", digitalocean_droplet.node.*.ipv4_address)}"` results in a comma-separated list containing all ip addresses, but since there is no `split` function, I have no way to extract individual IP addresses from this output in a later stage.

The documentation does not seem to address this issue. I'm open for suggestions on how to do this :)

**Comment by knuckolls** [comment]
fwiw we've run into this exact problem as well. Specifically due to the count functionality it'd be nice to be able to have array types in module outputs.

**Comment by mitchellh** [comment]
This will be fixed by #885

**Comment by mitchellh** [comment]
Fixed by #885

**Comment by thatha** [comment]
This seems to be ocurring again:

This

```
output "addresses_join" {
  value = ["${join(",", aws_instance.ssl_nginx.*.public_ip)}"]
}

output "addresses_star" {
  value = ["${aws_instance.ssl_nginx.*.public_ip}"]
}
```

produces this:

```
Outputs:

  addresses_join = 1.2.3.4,1.2.3.5
  addresses_star = 1.2.3.4
```
